### PR TITLE
Thosam qr code

### PR DESCRIPTION
**Pull request by Thosam1**
Implemented QR code
QRcodeProfileActivity - opens up a qr code for the user profile
QRcodeEventActivity - opens up a qr code for the event
QRcodeActivity - opens up a scanner

### CODE REVIEW

File: app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeActivity.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.journeyapps.barcodescanner.ScanContract;
+import com.journeyapps.barcodescanner.ScanIntentResult;
+import com.journeyapps.barcodescanner.ScanOptions;
+import com.sdpteam.connectout.R;
+
+public class QRcodeActivity extends AppCompatActivity {
+
+    private Button btn_scan;
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeActivity.java:18]: camelCase

File: app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeActivity.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        ScanOptions options = new ScanOptions();
+        options.setPrompt("Volum up to flash on");
+        options.setBeepEnabled(true);
+        options.setOrientationLocked(true);
+        options.setCaptureActivity(CaptureAct.class);
+
+        barLauncher.launch(options);
+    }
+
+    void handleScanResult(String resultText) {
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeActivity.java:44]: private missing

File: app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeEventActivity.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import android.view.View;
+import android.widget.Button;
+
+import com.sdpteam.connectout.R;
+
+public class QRcodeEventActivity extends AppCompatActivity {
+
+    private Button btn_show_qr_code;
+
+    private final String PROFILE_BASE_URI = "https://connect-out.com/events/";
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeEventActivity.java:20]: static?

File: app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeEventActivity.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+
+import com.sdpteam.connectout.R;
+
+public class QRcodeEventActivity extends AppCompatActivity {
+
+    private Button btn_show_qr_code;
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeEventActivity.java:18]: camelCase

File: app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeEventActivity.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+
+import com.sdpteam.connectout.R;
+
+public class QRcodeEventActivity extends AppCompatActivity {
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeEventActivity.java:16]: From what I understood this is an activity to create a QR code which is only used for the demo. Maybe you should add a TODO to remove it on next sprint when we will link the QR code to the event view

File: app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeModalActivity.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        qrCodeImageView = findViewById(R.id.qrcode_image);
+        modalTitle = findViewById(R.id.modal_title);
+
+        // retrieving data from intent
+        String title = getIntent().getStringExtra("title");
+        String qrCodeData = getIntent().getStringExtra("qrCodeData");
+
+        // displaying the title and the qr code
+        modalTitle.setText(title);
+        QRcodeGenerator qrCode = new QRcodeGenerator(); // any other way ?
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeModalActivity.java:33]: looks good to me... what other way are you thinking of ?

File: app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeModalActivity.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+        Button closeButton = findViewById(R.id.close_button);
+        closeButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                finish();
+            }
+        });
+    }
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeModalActivity.java:49]: You can simplify with lambda function ;)

File: app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeActivity.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        if (resultText != null) {
+            showDialog(resultText);
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeActivity.java:46]: At this point you should parse the resultText in order to create an intent to open the event or profile activity

File: app/src/main/java/com/sdpteam/connectout/qr_code/QRCodeScanner.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.widget.Toast;
+
+import com.google.zxing.client.android.Intents;
+import com.journeyapps.barcodescanner.CaptureActivity;
+import com.journeyapps.barcodescanner.ScanOptions;
+
+public class QRCodeScanner {
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/qr_code/QRCodeScanner.java:12]: Looks like this class is not used...

File: app/src/main/java/com/sdpteam/connectout/authentication/GoogleLoginActivity.java [hunk 10]
```diff
@@ -2,6 +2,7 @@
 
 import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract;
 import com.sdpteam.connectout.R;
+import com.sdpteam.connectout.profile.ProfileActivity;
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/authentication/GoogleLoginActivity.java:5]: Unused import, can you revert?

File: app/src/androidTest/java/com/sdpteam/connectout/qr_code/QRcodeActivityTest.java [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+//        activityScenarioRule.getScenario().onActivity(activity -> {
+//            String scanResult = "test result";
+//            activity.handleScanResult(scanResult);
+//            // Check that the dialog is shown with the correct message
+//            onView(withText(scanResult)).inRoot(isDialog()).check(matches(isDisplayed()));
+//            onView(withText("OK")).inRoot(isDialog()).perform(click());
+//        });
+//    }
+}
+
```
Comment (MikaelKalajdzic) [inline app/src/androidTest/java/com/sdpteam/connectout/qr_code/QRcodeActivityTest.java:121]: // Beautiful tests XD

File: app/src/androidTest/java/com/sdpteam/connectout/qr_code/QRcodeGeneratorTest.java [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    @Test
+    public void testGenerateQRCodeWithData() throws IllegalAccessException {
+        String data = "https://connect-out.com/events/0123456789";
+        Bitmap bitmap = qrCode.generateQRCode(data);
+        assertNotNull(bitmap);
+    }
+
+    //@Test(expected = IllegalAccessException.class)
+    public void testGenerateQRCodeWithNullData() throws IllegalAccessException {
+        qrCode.generateQRCode(null);
```
Comment (MikaelKalajdzic) [inline app/src/androidTest/java/com/sdpteam/connectout/qr_code/QRcodeGeneratorTest.java:28]: maybe use assertThrows() to check exceptions

File: app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeEventActivity.java (hunk 5 above)
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeEventActivity.java:16]: From what I understood this activity is only used for playing around and for the demo. Maybe you should add a TODO to remind that this has to be removed soon

**Review by MikaelKalajdzic — CHANGES_REQUESTED** [review]
good overall 
- Please create an intent after parsing the data from the QR code so that this connects to the event activity
- Improve the unit test

File: app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeModalActivity.java (hunk 6 above)
Comment (Thosam1) [reply app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeModalActivity.java:33]: nothing much, I initially thought about making this class static but it's alright now

**Review by MikaelKalajdzic — APPROVED** [review]

**Pull request merged**